### PR TITLE
ci: update site-deploy.yml

### LINKED ISSUE #39303: Deploy site when release version
类似 https://github.com/ant-design/ant-design-mobile/commit/72b8225153c4f4d23fa2c81f48b61d3a320d16eb#commitcomment-91035744

地址：https://ant-design-{version}.surge.sh

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Other (about what?)

### 🔗 Related issue link

close: #39303

### 💡 Background and solution

- Modify trigger events
- Add Surge Deployment
- Add Deployment Success Comment function

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: .github/workflows/doc-site.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# ref: https://github.com/ant-design/ant-design-mobile/blob/72b8225153c4f4d23fa2c81f48b61d3a320d16eb/.github/workflows/doc-site.yml
+
+name: Doc Site
+on:
+  release:
+    types: [published]
```
Comment (MadCcc) [inline .github/workflows/doc-site.yml:6]: 监听 tag 好点？

Comment (Wxh16144) [reply .github/workflows/doc-site.yml:6]: 嗯嗯, 我修改一下。
不过下面的 steps 我只测试了后半部分，前半部分调试赐教一下?

**Comment by afc163** [comment]
加到 https://github.com/ant-design/ant-design/blob/master/.github/workflows/site-deploy.yml 吧，我在这个 PR 加了标签不会自动关闭了。

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/blob/master/.github/workflows/site-deploy.yml 感觉也实现的不太好，on create 然后再去判断类型，这样所有操作都会触发这个 action，不太必要。

**Review by MadCcc — APPROVED** [review]

File: .github/workflows/site-deploy.yml [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
+  release:
+    types: [published]
 
 permissions:
   contents: write
 
 jobs:
   setup:
     runs-on: ubuntu-latest
-    if: github.event.ref_type == 'tag' && (contains(github.event.ref, '-') == false)
```
Comment (MadCcc) [inline .github/workflows/site-deploy.yml:13]: (contains(github.event.ref, '-') == false)
这个还是要的，不然测试版本也会部署站点

Comment (Wxh16144) [reply .github/workflows/site-deploy.yml:13]: 噢对，测试版的 tag。 v5.0.0-rc、v5.0.0-beta.1 等这些

**Review by afc163 — APPROVED** [review]

**Pull request merged**